Compilation fails when passing `vulkan="no"`

### Tested versions

Tested on current master (b7feebe)

### System information

Windows 10, x86_64, MSVC 14.3

### Issue description

Build terminates when excluding vulkan.
```
> scons p=windows
scons: Reading SConscript files ...
Auto-detected 6 CPU cores available for build parallelism. Using 5 cores by default. You can override it with the -j argument.
Using SCons-detected MSVC version 14.3, arch x86_64
Building for platform "windows", architecture "x86_64", target "template_release".

...

platform\windows\display_server_windows.cpp(3411): error C2065: 'rendering_device': undeclared identifier
platform\windows\display_server_windows.cpp(3411): error C2065: 'rendering_device': undeclared identifier
scons: *** [platform\windows\display_server_windows.windows.template_release.x86_64.obj] Error 2
scons: building terminated because of errors.
[Time elapsed: 00:00:15.96]
```

### Steps to reproduce

See above

### Minimal reproduction project (MRP)

N/A

--- a/platform/linuxbsd/x11/display_server_x11.cpp
+++ b/platform/linuxbsd/x11/display_server_x11.cpp
@@ -5201,10 +5201,11 @@ bool DisplayServerX11::is_window_transparency_available() const {
 	if (XGetSelectionOwner(x11_display, net_wm_cm) == None) {
 		return false;
 	}
-
+#if defined(RD_ENABLED)
 	if (rendering_device && !rendering_device->is_composite_alpha_supported()) {
 		return false;
 	}
+#endif
 	return OS::get_singleton()->is_layered_allowed();
 }
 

--- a/platform/macos/display_server_macos.mm
+++ b/platform/macos/display_server_macos.mm
@@ -3315,9 +3315,11 @@ void DisplayServerMacOS::delete_status_indicator(IndicatorID p_id) {
 }
 
 bool DisplayServerMacOS::is_window_transparency_available() const {
+#if defined(RD_ENABLED)
 	if (rendering_device && !rendering_device->is_composite_alpha_supported()) {
 		return false;
 	}
+#endif
 	return OS::get_singleton()->is_layered_allowed();
 }
 

--- a/platform/windows/display_server_windows.cpp
+++ b/platform/windows/display_server_windows.cpp
@@ -3408,9 +3408,11 @@ bool DisplayServerWindows::is_window_transparency_available() const {
 			return false;
 		}
 	}
+#if defined(RD_ENABLED)
 	if (rendering_device && !rendering_device->is_composite_alpha_supported()) {
 		return false;
 	}
+#endif
 	return OS::get_singleton()->is_layered_allowed();
 }
 

